build/storage/scripts/disk_infrastructure.py

```diff
@@ -0,0 +1,207 @@
+    cmd = 'lsblk --output "NAME"'
+    out = socket_functions.send_command_over_unix_socket(
+        sock=sock, cmd=cmd, wait_for_secs=1
+    )
+    number_of_virtio_blk_devices = len(re.findall("vd", out))
+    return number_of_virtio_blk_devices
+
+
+def is_virtio_blk_attached(sock: str) -> int:
+    if get_number_of_virtio_blk(sock) == 0:
+        logging.error("virtio-blk is not found")
+        return 1
+    logging.info("virtio-blk is found")
+    return 0
+
+
+def is_virtio_blk_not_attached(sock: str) -> int:
+    if is_virtio_blk_attached(sock):
+        return 0
+    return 1
+
+
+def check_number_of_virtio_blk_devices(
+    vm_serial: str, expected_number_of_devices: int
+) -> int:
+    number_of_devices = get_number_of_virtio_blk(vm_serial)
+    if number_of_devices != expected_number_of_devices:
+        logging.error(
+            f"Required number of devices '{expected_number_of_devices}' does "
+            f"not equal to actual number of devices '{number_of_devices}'"
+        )
+        return 1
+    else:
+        logging.info(f"Number of attached virtio-blk devices is '{number_of_devices}'")
+        return 0
+
+
+def create_and_expose_subsystem_over_tcp(
+    ip_addr: str, nqn: str, port_to_expose: str, storage_target_port: int
+) -> None:
+    requests = [
+        {
+            "method": "nvmf_create_subsystem",
+            "params": {
+                "nqn": nqn,
+                "serial_number": "SPDK00000000000001",
+                "allow_any_host": True,
+                "max_namespaces": 1024,
+            },
+        },
+        {
+            "method": "nvmf_create_transport",
+            "params": {"trtype": "TCP", "io_unit_size": 8192},
+        },
+        {
+            "method": "nvmf_subsystem_add_listener",
+            "params": {
+                "nqn": nqn,
+                "listen_address": {
+                    "trtype": "TCP",
+                    "adrfam": "IPv4",
+                    "traddr": ip_addr,
+                    "trsvcid": port_to_expose,
+                },
+            },
+        },
+    ]
+    for request in requests:
+        send_rpc_request(
+            request=request,
+            addr=ip_addr,
+            port=storage_target_port,
+        )
+
+
+def create_ramdrive_and_attach_as_ns_to_subsystem(
+    ip_addr: str,
+    ramdrive_name: str,
+    ramdrive_size_in_mb: int,
+    nqn: str,
+    storage_target_port: int,
+) -> str:
+    requests = [
+        {
+            "method": "bdev_malloc_create",
+            "params": {
+                "name": ramdrive_name,
+                "num_blocks": ramdrive_size_in_mb * 1024 * 1024 // 512,
+                "block_size": 512,
+            },
+        },
+        {
+            "method": "nvmf_subsystem_add_ns",
+            "params": {"nqn": nqn, "namespace": {"bdev_name": ramdrive_name}},
+        },
+        {"method": "bdev_get_bdevs", "params": {"name": ramdrive_name}},
+    ]
+    for request in requests:
```

Comment:
this code is common for several functions - you may consider writing an utility for that

Reply:
I've made a utility for that - another function send_requests()